### PR TITLE
Automatically update nodePackages

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Motivation for this change

Currently running at https://github.com/SuperSandro2000/nixpkgs/runs/1883519163?check_suite_focus=true

Example PR https://github.com/SuperSandro2000/nixpkgs/pull/2/

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FRidh** [comment]
I wonder whether we should extend the concept of `updateScript` also to package sets.

**Comment by SuperSandro2000** [comment]
> I wonder whether we should extend the concept of `updateScript` also to package sets.

Might be an option but I lack the nix knowledge to implement that.

**Comment by happysalada** [comment]
I love the idea!

**Comment by zowoq** [comment]
Probably want to notify people when the PR is created, could cc them in the PR comment or add ` pkgs/development/node-packages/node-packages.nix ` to CODEOWNERS.

A failure notification like we have for [`merge-staging`](https://github.com/NixOS/nixpkgs/blob/b5c9a550b96b67d1f3338515fe826442f5d62881/.github/workflows/merge-staging.yml#L32-L40) may also be helpful.

**Comment by SuperSandro2000** [comment]
> Probably want to notify people when the PR is created, could cc them in the PR comment or add pkgs/development/node-packages/node-packages.nix to CODEOWNERS.

Who would we ping? I couldn't find any real code owner except maybe @mic92.

> A failure notification like we have for merge-staging may also be helpful.

That would be helpful. Should I use the issue number of this PR or create a new issue which is just for failures?

**Comment by happysalada** [comment]
Maybe there is a need for a node team ?
Just thinking out loud here.

**Comment by FRidh** [comment]
At the very least we need codeowners for node-packages.nix.

**Comment by happysalada** [comment]
One problem i see with this is that if the update of one package fails, the whole PR fails. If there was a way to do it per package, it would be easier to merge the ones that did not fail.

**Comment by SuperSandro2000** [comment]
> One problem i see with this is that if the update of one package fails, the whole PR fails. If there was a way to do it per package, it would be easier to merge the ones that did not fail.

There is no such concept right now and I don't know if it would be doable.

> At the very least we need codeowners for node-packages.nix.

I would suggest myself but I someone or two more wouldn't hurt.

**Comment by happysalada** [comment]
I agree with you, I don't know how to do it either.

I would be interested too in being in the nodePackages maintainer if you create a team.

**Comment by happysalada** [comment]
Hey, a crazy idea.

What about keeping each package in a separate directory (with the same layout as the current directory). Or maybe groups of packages that belong together.
That way packages or group of packages could be updated and added independently.
In the top directory we could reference the packages from their directory.

Let me know if it doesn't make sense.

**Comment by happysalada** [comment]
To favor discussion, I have created a PR with what I have in mind
https://github.com/NixOS/nixpkgs/pull/113294
I would be curious to get some feedback, if anybody has time.
I think it's still early stage, but it's just to put forward an idea.

File: .github/workflows/update-node-packages.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        with:
+          commit-message: Update nodePackages
+          committer: GitHub <noreply@github.com>
+          author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+          signoff: false
+          branch: update-node-packages
+          delete-branch: true
+          title: 'WIP: Update nodePackages'
+          body: |
+            Automated weekly nodePackages update
```
Comment (andir) [inline .github/workflows/update-node-packages.yml:27]: Please add a link to the script that runs this automation in the commit message / the PR message.

**Comment by FRidh** [comment]
Could you add also an entry in https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS

**Comment by SuperSandro2000** [comment]
Added me to codeowners and the path to the script in the commit title and body.
I currently do not want to create a team for node.

**Comment by doronbehar** [comment]
I am too in favor of automated updates to all node packages but I never thought it would be acceptable in Nixpkgs, I also don't think this should go through the PR bottle neck as suggested here but I understand why it's a sensitive issue in Nixpkgs. Hence I created:

https://gitlab.com/doronbehar/nixNodeJSPkgs

Which is a flake that runs automatic updates via GitLab's CI and commits them directly. It's still not ready for an official announcement yet, because it fails to build the only package I'm interested in due to https://github.com/svanderburg/node2nix/issues/233 .

I also want the update to run every day, because the package I wish to build there gets a new release every 1-2 days: https://github.com/balena-io/balena-cli/releases .

**Comment by Mic92** [comment]
Would be still useful to have this.

**Comment by SuperSandro2000** [comment]
Whenever node-packages is a problem for you, the package should be converted to ``buildNpmPackage``.

**Pull request closed without merging**